ipywidgets_bokeh/widget.py

```diff
@@ -5,24 +5,69 @@
 # The full license is in the file LICENSE.txt, distributed with this software.
 #-----------------------------------------------------------------------------
 
-from bokeh.core.properties import Any
+from bokeh.core.properties import Any, String
 from bokeh.models.layouts import HTMLBox
 
 from ipywidgets import embed, Widget
 
 from .kernel import kernel
 
+
 class IPyWidget(HTMLBox):
+    """Wrap an IPyWidget for embedding in a bokeh app.
+
+    Parameters
+    ----------
+    widget : Widget
+        ipywidget to wrap for embedding in a bokeh app
+    **kwargs
+        All kwargs are passed to bokeh.model.Model except 'cdn', if present.
+
+        The CDN for external JS resources uses the 'cdn' kwarg if set; otherwise use
+        self.default_cdn. After initialization (but before rendering), the CDN can be
+        set using IPyWidget.set_cdn; see below.
+
+    """
 
     __javascript__ = [
         "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js",
     ]
 
     bundle = Any()
+    cdn = String()
+    default_cdn = 'https://unpkg.com'
 
     def __init__(self, *, widget: Widget, **kwargs):
+        cdn = kwargs.pop('cdn', self.default_cdn)
+
         super().__init__(**kwargs)
         spec = widget.get_view_spec()
         state = Widget.get_manager_state(widgets=[])
         state["state"] = embed.dependency_state([widget], drop_defaults=True)
-        self.bundle = dict(spec=spec, state=state)
+        self.bundle = {'spec': spec, 'state': state}
+        self.cdn = cdn
+
+    def set_cdn(self, cdn: str):
+        """Set the CDN used for this widget.
+
+        Parameters
+        ----------
+        cdn : str
+            CDN to use for loading external JS resources if necessary; defaults to
+            'https://unpkg.com'
+        """
+        self.cdn = cdn
+
+    @classmethod
+    def set_default_cdn(cls, cdn: str):
+        """Set the default CDN used across all ipywidgets_bokeh.widget.IPyWidget instances.
+
+        Can be overridden by setting self.cdn.
+
+        Parameters
+        ----------
+        cdn : str
+            CDN to use for loading external JS resources if necessary; defaults to
+            'https://unpkg.com'
+        """
+        cls.default_cdn = cdn
```

Comment:
I'm not sure we really need these. Setting attributes seems like a sufficiently simple API

Reply:
Yep, agreed. I've removed these two functions.